Detectors/MUON/MCH/Mapping/README.md

```diff
@@ -47,12 +47,60 @@ Note that cathode-specific segmentations can be retrieved from `Segmentation`
 objects (using bending() and nonBending() methods) or created from scratch
 using the `CathodeSegmentation(int detElemId, bool isBendingPlane)` constructor.
 
+## The segmentation() utility function
+
+To create segmentations and avoid duplicating objects, you can use the `segmentation`
+ function (mind the lower case first letter s).
+
+For instance :
+
+```cpp
+const Segmentation& seg1 = segmentation(501);
+...
+(some code here)
+...
+auto& seg2 = segmentation(501);
+```
+
+will result in only one object for segmentation 501 to be created.
+
+On the contrary, if you use :
+
+```cpp
+Segmentation seg1{501};
+...
+...
+Segmentation seg2{501};
+```
+
+two separate objects will be created (and you will be hit twice by the
+object creation time).
+
+> Note that depending on your use case, *both* approaches might be valid ones !
+> While the factory one ensures you get only get one object for each detection
+> element, it creates all detection elements at once. So, if you only need the
+> segmentation of one detection element, you'd better off *not* using the
```

Comment:
Why not implement the factory as a std::map-like container that is filled dynamically whenever a new detection element is requested?

Reply:
A map has a search cost at each access. The idea behind the factory building the full mapping is the assumption that the most likely access patterns in the end are either one segmentation "per process" (in which case you don't need the factory function at all) or all of them. If that assumption turns out to be wrong then we'll adapt in due time ;-)